File: native/tests/fixtures/sample.py
"""Tiny fixture used by parse_python integration tests."""

import os
from typing import List


def greet(name):
    return f"hello {name}"


class Greeter:
    def hello(self, name):
        return greet(name)

    def shout(self, name):
        return self.hello(name).upper()


def main():
    g = Greeter()
    return g.shout("world")

File: native/tests/fixtures/sample.rs
//! Tiny fixture used by parse_rust integration tests.

pub struct Greeter {
    pub prefix: String,
}

impl Greeter {
    pub fn new(prefix: &str) -> Self {
        Self {
            prefix: prefix.to_string(),
        }
    }

    pub fn hello(&self, name: &str) -> String {
        format!("{} {}", self.prefix, name)
    }

    pub fn shout(&self, name: &str) -> String {
        self.hello(name).to_uppercase()
    }
}

pub fn greet(name: &str) -> String {
    format!("hello {}", name)
}

pub fn run() -> String {
    let g = Greeter::new("hi");
    g.shout("world")
}
